### PR TITLE
feat: ✨ Add runtime API to Proofs Dealer pallet

### PR DESCRIPTION
**Pull request by ffarall**
In this PR:

1. Add runtime API to Proofs Dealer pallet that enables the SH node to:
    1. Get the last tick for which a Provider submitted a proof.
    2. Get the last tick at which there was a checkpoint challenge.
    3. Get the last checkpoint challenges.

### CODE REVIEW

File: pallets/proofs-dealer/src/utils.rs [hunk 1]
```diff
@@ -691,3 +694,39 @@ impl<T: pallet::Config> ProofsDealerInterface for Pallet<T> {
...
+
+    pub fn get_last_checkpoint_challenge_tick() -> BlockNumberFor<T> {
+        LastCheckpointTick::<T>::get()
+    }
+
+    pub fn get_checkpoint_challenges(
+        tick: BlockNumberFor<T>,
+    ) -> Result<Vec<(KeyFor<T>, Option<TrieRemoveMutation>)>, GetCheckpointChallengesError> {
+        // Check that the tick is smaller than the last checkpoint tick.
+        if LastCheckpointTick::<T>::get() >= tick {
```
Comment (links234) [inline pallets/proofs-dealer/src/utils.rs:723]: Shouldn't this be `LastCheckpointTick::<T>::get() < tick` ?

**Review by links234 — CHANGES_REQUESTED** [review]
Besides a small bug, looks good!

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -522,6 +521,26 @@ impl_runtime_apis! {
...
+        fn query_earliest_file_volunteer_block(bsp_id: Hash, file_key: H256) -> Result<BlockNumber, QueryFileEarliestVolunteerBlockError> {
+            FileSystem::query_earliest_file_volunteer_block(bsp_id, file_key)
+        }
+    }
+
+    impl pallet_proofs_dealer_runtime_api::ProofsDealerApi<Block, ProviderIdFor<Runtime>, BlockNumber, KeyFor<Runtime>, TrieRemoveMutation> for Runtime {
+        fn get_last_tick_provider_submitted_proof(provider_id: &ProviderIdFor<Runtime>) -> Result<BlockNumber, GetLastTickProviderSubmittedProofError> {
+            ProofsDealer::get_last_tick_provider_submitted_proof(provider_id)
+        }
+        fn get_last_checkpoint_challenge_tick() -> BlockNumber {
```
[the remark this answers is not in the document]
Comment (ffarall) [reply runtime/src/lib.rs:535]: Yeah, good point

**Review by links234 — APPROVED** [review]
LGTM!

**Pull request merged**